packages/cli/src/lib/setup/setupBackup.ts

```diff
@@ -679,18 +679,23 @@ export class BackupRestore {
         // replace all hostnames of instances etc with the new host
         data = data.replace(/\$\$__hostname__\$\$/g, hostname);
         fs.writeFileSync(`${tmpDir}/backup/backup_.json`, data);
-        let restore;
+        let restore: BackupObject;
         try {
             restore = JSON.parse(data);
         } catch (err) {
             console.error(`Cannot parse "${tmpDir}/backup/backup_.json": ${err.message}`);
             return EXIT_CODES.CANNOT_RESTORE_BACKUP;
         }
 
+        if (!restore.objects) {
+            console.error('The backup does not contain any objects.');
+            return EXIT_CODES.CANNOT_RESTORE_BACKUP;
+        }
+
         // check that the same controller version is installed as it is contained in backup
         const exitCode = this._ensureCompatibility(
             controllerDir,
-            restore.config ? restore.config.system.hostname || hostname : hostname,
+            restore.config?.system ? restore.config.system.hostname || hostname : hostname,
```

Comment:
```suggestion
            restore.config?.system?.hostname || hostname,
```

Reply:
not sure if `restore.config.system.hostname` can be empty. If not, I'd use `??` in place of `||`

Reply:
If empty string, the variable `hostname` should be applied, as it is done currently, so your suggestion makes sense to simplify it.